src/Http/Http.Abstractions/src/PathString.cs

```diff
@@ -99,10 +94,15 @@ private static string ToEscapedUriComponent(string value, int i)
         var count = i;
         var requiresEscaping = false;
 
-        while (i < value.Length)
+        while (true)
         {
-            var isPercentEncodedChar = PathStringHelper.IsPercentEncodedChar(value, i);
-            if (PathStringHelper.IsValidPathChar(value[i]) || isPercentEncodedChar)
+            if ((uint)i >= (uint)value.Length)
```

Comment:
Why not `while ((uint)i < (uint)value.Length)`?

Reply:
Or even the original `while (i < value.Length)`? Is the cast buying us anything?

Reply:
The current pattern is something that the JIT is able to recognize and elide the range check on the `value[i]` access below.
`while ((uint)i < (uint)value.Length)` is something that should [eventually](https://github.com/dotnet/runtime/issues/83349) get you the same result.

Reply:
Stick with the expected code, let the JIT take care of its own issues.